GliaWidgets/Coordinator/Engagement/EngagementCoordinator.swift

```diff
@@ -0,0 +1,274 @@
+import UIKit
+
+final class EngagementCoordinator: UIViewControllerCoordinator {
+    enum DelegateEvent {
+        case minimize
+        case engaged(operatorImageUrl: String?)
+        case started
+        case ended
+        case engagementChanged(EngagementKind)
+        case unreadMessagesCount(Int)
+    }
+
+    var delegate: ((DelegateEvent) -> Void)?
+    var engagementKind: EngagementKind {
+        didSet {
+            delegate?(.engagementChanged(engagementKind))
+        }
+    }
+
+    private let unreadMessages = ObservableValue<Int>(with: 0)
+    private let screenShareHandler = ScreenShareHandler()
+
+    private weak var presenter: UINavigationController?
+
+    private let interactor: Interactor
+    private let isWindowVisible: ObservableValue<Bool>
+    private let viewFactory: ViewFactory
+    private let environment: Environment
+
+    init(
+        interactor: Interactor,
+        engagementKind: EngagementKind,
+        isWindowVisible: ObservableValue<Bool>,
+        viewFactory: ViewFactory,
+        environment: Environment
+    ) {
+        self.interactor = interactor
+        self.engagementKind = engagementKind
+        self.isWindowVisible = isWindowVisible
+        self.viewFactory = viewFactory
+        self.environment = environment
+
+        super.init(
+            environment: .init(
+                uuid: environment.uuid
+            )
+        )
+
+        setup()
+    }
+
+    override func start() -> UIViewController {
+        defer {
```

Comment:
why do we ignore `super.start()` here?

Reply:
super start would only call fatalerror, so in this instance the override without super is intentional for all uiviewcontrollercoordinator subclasses.